addons/account_edi/models/account_edi_format.py

```diff
@@ -499,21 +499,71 @@ def _retrieve_partner(self, name=None, phone=None, mail=None, vat=None):
         :param vat:     The vat number of the partner.
         :returns:       A partner or an empty recordset if not found.
         '''
-        domains = []
-        for value, domain in (
-            (name, [('name', 'ilike', name)]),
-            (phone, expression.OR([[('phone', '=', phone)], [('mobile', '=', phone)]])),
-            (mail, [('email', '=', mail)]),
-            (vat, [('vat', 'like', vat)]),
-        ):
-            if value is not None:
-                domains.append(domain)
-
-        domain = expression.AND([
-            expression.OR(domains),
-            [('company_id', 'in', [False, self.env.company.id])],
-        ])
-        return self.env['res.partner'].search(domain, limit=1)
+        def search_with_vat(extra_domain):
+            if not vat:
+                return None
+
+            # Sometimes, the vat is specified with some whitespaces.
+            normalized_vat = vat.replace(' ', '')
+
+            # Country prefix is optional.
+            country_prefix = normalized_vat[:2]
+            if country_prefix.isalpha():
+                normalized_vat = normalized_vat[2:]
```

Comment:
Can't we have false positives with that ? Some countries accept letters in their VAT number outside of the country code. Maybe we should first check with, then without (if we don't have any match) the first two characters if they are letters ?